tests/testthat/test_add_robust_constraints.R

```diff
@@ -1,19 +1,206 @@
+library(prioritizr)
+library(terra)
+
 test_that("compile (single zone)", {
-  # TODO
+  skip_if_not_installed("terra")
+  # import data
+  sim_pu_raster <- prioritizr::get_sim_pu_raster()
+  sim_features <- prioritizr::get_sim_features()
+  # define feature groupings
+  x <- rep_len(c("a", "b"), terra::nlyr(sim_features))
+
+  conf_level <- 0.9
+  # build problem
+  p <-
+    prioritizr::problem(sim_pu_raster, sim_features) |>
+    add_robust_min_set_objective(method = "Chance") |>
+    prioritizr::add_relative_targets(0.1) |>
+    add_constant_robust_constraints(x, conf_level) |>
+    prioritizr::add_binary_decisions()
+
+  cp <- compile(p)
+
+  group_cardinality <- unname(table(x))
+
+  A_row_sum <- apply(cp$A(), 1, sum)
+
+  # Expect the length of rhs to be equal to 7
+  expect_equal(length(cp$rhs()), 7)
+
+  # Expect the rhs length to be equal to the second dimension of A
+  expect_equal(nrow(cp$A()), length(cp$rhs()))
+
+  # Expect the row sums to be less than the cardinality of the group
+  expect_true(all((group_cardinality - A_row_sum[6:7]) > -1e-5))
+
+  # Expect the confidence level to be reflected in the RHS
+  expect_true(all((group_cardinality*(1-conf_level) - cp$rhs()[6:7]) > 1e-5))
 })
 
 test_that("solve (single zone)", {
-  # TODO
+  skip_if_not_installed("terra")
+  # import data
+  sim_pu_raster <- prioritizr::get_sim_pu_raster()
+  sim_features <- prioritizr::get_sim_features()
```

Comment:
rather than using the simulated data, here in the solve tests, i suggest manually creating some simulated data where there is a single obvious optimal solution, and then solving the problem to see if it returns the obvious optimal solution.